tpu-client/src/nonblocking/tpu_client.rs

```diff
@@ -760,9 +726,43 @@ impl LeaderTpuService {
 
         let recent_slots = RecentLeaderSlots::new(start_slot);
         let slots_in_epoch = rpc_client.get_epoch_info().await?.slots_in_epoch;
-        let leaders = rpc_client
-            .get_slot_leaders(start_slot, LeaderTpuCache::fanout(slots_in_epoch))
-            .await?;
+
+        // When a cluster is starting, we observe an invalid slot range failure that goes away after a
+        // retry. It seems as if the leader schedule is not available, but it should be. The logic
+        // below retries the RPC call in case of an invalid slot range error.
+        let tpu_leader_service_creation_timeout = Duration::from_secs(20);
+        let retry_interval = Duration::from_secs(1);
+        let leaders = timeout(tpu_leader_service_creation_timeout, async {
+            loop {
+                // TODO: The root cause appears to lie within the `rpc_client.get_slot_leaders()`.
+                // It might be worth debugging further and trying to understand why the RPC
+                // call fails. There may be a bug in the `get_slot_leaders()` logic or in the
+                // RPC implementation
+                match rpc_client
+                    .get_slot_leaders(start_slot, LeaderTpuCache::fanout(slots_in_epoch))
+                    .await
+                {
+                    Ok(leaders) => return Ok(leaders),
+                    Err(client_error) => {
+                        if is_invalid_slot_range_error(&client_error) {
+                            sleep(retry_interval).await;
+                            continue;
+                        } else {
+                            return Err(client_error);
+                        }
+                    }
+                }
+            }
+        })
+        .await
+        .map_err(|_| {
+            TpuSenderError::Custom(format!(
+                "Failed to get slot leaders connecting to: {}, timeout: {}s. Invalid slot range",
+                websocket_url,
+                tpu_leader_service_creation_timeout.as_secs()
```

Comment:
`Duration` has a pretty good `Debug` printer:

```
Duration::from_secs(1):      1s
Duration::from_millis(1500): 1.5s
Duration::from_millis(10):   10ms
Duration::from_micros(1500): 1.5ms
```
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=f2689b805a9edeb24d6e277063c6129d

```suggestion
                "Failed to get slot leaders connecting to: {}, timeout: {:?}s. Invalid slot range",
                websocket_url,
                tpu_leader_service_creation_timeout
```

P.S. Essentially, `as_secs()` is almost always a bad choice, if you want to print the value.
Either `Debug` or [`humantime`](https://docs.rs/humantime/latest/humantime) are strictly better.